### PR TITLE
When removing ImageRepository, check if there is another ImageRepository

### PR DESCRIPTION
**Pull request by rcerven**
for the same repository, if there is, don't remove repository but only robot accounts for ImageRepository which is to be removed

[STONEBLD-2876](https://issues.redhat.com//browse/STONEBLD-2876)

### CODE REVIEW

File: controllers/imagerepository_controller.go [hunk 1]
```diff
@@ -118,8 +118,27 @@ func (r *ImageRepositoryReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		}
 
 		if controllerutil.ContainsFinalizer(imageRepository, ImageRepositoryFinalizer) {
+			// Check if there isn't other ImageRepository for the same repository from other component
```
Comment (mmorhun) [inline controllers/imagerepository_controller.go:121]: I think it could be a separate func

File: controllers/imagerepository_controller.go [hunk 2]
```diff
@@ -918,3 +928,26 @@ func generateDockerconfigSecretData(quayImageURL string, robotAccount *quay.Robo
...
+
+	imageRepositoryUrl := imageRepository.Status.Image.URL
+	imageRepositoryName := imageRepository.ObjectMeta.Name
+	imageRepositoryFound := false
+	for _, imageRepo := range imageRepositoriesList.Items {
+		if imageRepositoryUrl == imageRepo.Status.Image.URL {
+			if imageRepositoryName == imageRepo.ObjectMeta.Name {
+				continue
+			}
+			imageRepositoryFound = true
```
Comment (mmorhun) [inline controllers/imagerepository_controller.go:948]: We don't need `imageRepositoryFound` variable at all, just return here

File: controllers/imagerepository_controller.go [hunk 3]
```diff
@@ -918,3 +928,26 @@ func generateDockerconfigSecretData(quayImageURL string, robotAccount *quay.Robo
...
+		return false, err
+	}
+
+	imageRepositoryUrl := imageRepository.Status.Image.URL
+	imageRepositoryName := imageRepository.ObjectMeta.Name
+	imageRepositoryFound := false
+	for _, imageRepo := range imageRepositoriesList.Items {
+		if imageRepositoryUrl == imageRepo.Status.Image.URL {
+			if imageRepositoryName == imageRepo.ObjectMeta.Name {
+				continue
```
Comment (mmorhun) [inline controllers/imagerepository_controller.go:946]: Would be good to have a comment here, like `Skipping original object`

File: controllers/imagerepository_controller.go (hunk 2 above)
Comment (rcerven) [reply controllers/imagerepository_controller.go:948]: done

File: controllers/imagerepository_controller.go (hunk 3 above)
Comment (rcerven) [reply controllers/imagerepository_controller.go:946]: done

**Review by mmorhun — APPROVED** [review]

**Review by mkosiarc — APPROVED** [review]

**Pull request merged**